File: app/src/main/java/com/qiscus/sdksample/qiscussamplesdk/ContactAdapter.java
package com.qiscus.sdksample.qiscussamplesdk;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

public class ContactAdapter extends RecyclerView.Adapter<ContactAdapter.ContactVH> {

    private Context context;
    private List<Contact> contacts;
    private OnClickListener clickListener;

    public ContactAdapter(Context context, List<Contact> contacts) {
        this.context = context;
        if (contacts != null) {
            this.contacts = contacts;
        } else {
            this.contacts = new ArrayList<>();
        }
    }

    @Override
    public ContactVH onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ContactVH(LayoutInflater.from(context).inflate(R.layout.item_contact, parent, false), clickListener);
    }

    @Override
    public void onBindViewHolder(ContactVH holder, int position) {
        holder.bind(contacts.get(position));
    }

    @Override
    public int getItemCount() {
        return contacts.size();
    }

    public List<Contact> getContacts() {
        return contacts;
    }

    public void setOnClickListener(OnClickListener clickListener) {
        this.clickListener = clickListener;
    }

    public class ContactVH extends RecyclerView.ViewHolder implements View.OnClickListener {

        private TextView nameText;
        private OnClickListener clickListener;

        public ContactVH(View itemView, OnClickListener clickListener) {
            super(itemView);
            nameText = (TextView) itemView.findViewById(R.id.name);
            this.clickListener = clickListener;

            itemView.setOnClickListener(this);
        }

        public void bind(Contact contact) {
            nameText.setText(contact.getName());
        }

        @Override
        public void onClick(View v) {
            if (clickListener != null) {
                clickListener.onClick(getAdapterPosition());
            }
        }
    }

    public interface OnClickListener {
        void onClick(int position);
    }
}

File: app/src/main/java/com/qiscus/sdksample/qiscussamplesdk/ContactAdapter.kt
package com.qiscus.sdksample.qiscussamplesdk

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import java.util.*

class ContactAdapter(private val context: Context, contacts: MutableList<Contact>?) : RecyclerView.Adapter<ContactAdapter.ContactVH>() {
    var contacts: MutableList<Contact>? = null
        private set
    private var clickListener: OnClickListener? = null

    init {
        if (contacts != null) {
            this.contacts = contacts
        } else {
            this.contacts = ArrayList<Contact>()
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContactVH {
        return ContactVH(LayoutInflater.from(context).inflate(R.layout.item_contact, parent, false), clickListener)
    }

    override fun onBindViewHolder(holder: ContactVH, position: Int) {
        holder.bind(contacts!![position])
    }

    override fun getItemCount(): Int {
        return contacts!!.size
    }

    fun setOnClickListener(clickListener: OnClickListener) {
        this.clickListener = clickListener
    }

    inner class ContactVH(itemView: View, private val clickListener: OnClickListener?) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        private val nameText: TextView = itemView.findViewById(R.id.name) as TextView

        init {
            itemView.setOnClickListener(this)
        }

        fun bind(contact: Contact) {
            nameText.text = contact.name
        }

        override fun onClick(v: View) {
            clickListener?.onClick(adapterPosition)
        }
    }

    interface OnClickListener {
        fun onClick(position: Int)
    }
}
